diff --git a/kernel/cgroup.c b/kernel/cgroup.c
index 267309e3..e87b9a3 100644
--- a/kernel/cgroup.c
+++ b/kernel/cgroup.c
@@ -2135,8 +2135,8 @@ int subsys_cgroup_allow_attach(struct cgroup *cgrp, struct cgroup_taskset *tset)
 	cgroup_taskset_for_each(task, cgrp, tset) {
 		tcred = __task_cred(task);
 
-		if (current != task && cred->euid != tcred->uid &&
-		    cred->euid != tcred->suid)
+		if (current != task && cred->euid.val != tcred->uid.val &&
+		    cred->euid.val != tcred->suid.val)
 			return -EACCES;
 	}
 
