diff --git a/net/netfilter/xt_quota2.c b/net/netfilter/xt_quota2.c
index 4328562..b598a93 100644
--- a/net/netfilter/xt_quota2.c
+++ b/net/netfilter/xt_quota2.c
@@ -231,7 +231,7 @@ q2_get_counter(const struct xt_quota_mtinfo2 *q)
 		spin_unlock_bh(&counter_list_lock);
 		goto out;
 	}
-	proc_set_user(p, quota_list_uid, quota_list_gid);
+	proc_set_user(p, KUIDT_INIT(quota_list_uid), KGIDT_INIT(quota_list_gid));
 	return e;
 
  out:
