diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index b16a184..824677c 100755
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -58,9 +58,9 @@ scm_version()
 		[ -w . ] && git update-index --refresh --unmerged > /dev/null
 
 		# Check for uncommitted changes
-		if git diff-index --name-only HEAD | grep -qv "^scripts/package"; then
-			printf '%s' -dirty
-		fi
+#		if git diff-index --name-only HEAD | grep -qv "^scripts/package"; then
+#			printf '%s' -dirty
+#		fi
 
 		# All done with git
 		return
